src/fix_graphics.cpp

```diff
@@ -0,0 +1,677 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   LAMMPS development team: developers@lammps.org
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "fix_graphics.h"
+
+#include "comm.h"
+#include "domain.h"
+#include "dump_image.h"
+#include "error.h"
+#include "input.h"
+#include "lattice.h"
+#include "math_extra.h"
+#include "memory.h"
+#include "modify.h"
+#include "update.h"
+#include "variable.h"
+
+#include <cstring>
+#include <utility>
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+enum { SPHERE, CYLINDER, ARROW, PROGBAR };
+enum { X = 0, Y, Z };
+
+/* ---------------------------------------------------------------------- */
+
+FixGraphics::FixGraphics(LAMMPS *lmp, int narg, char **arg) :
+    Fix(lmp, narg, arg), imgobjs(nullptr), imgparms(nullptr)
+{
+  if (narg < 4) utils::missing_cmd_args(FLERR, "fix graphics", error);
+
+  // parse mandatory arg
+
+  nevery = utils::inumeric(FLERR, arg[3], false, lmp);
+  if (nevery <= 0) error->all(FLERR, 3, "Illegal fix graphics nevery value {}", nevery);
+  global_freq = nevery;
+
+  numobjs = 0;
+  varflag = 0;
+
+  int iarg = 4;
+  while (iarg < narg) {
+    if (strcmp(arg[iarg], "sphere") == 0) {
```

Comment:
I understand arrows/progress bars, but can't one just draw a sphere/cylinder using the region command? Those two options seem redundant.

Reply:
Overkill in which manner? Is it clunky to add them to the dump image command (does a union work)? Or as in it's slow? Or something else?